cmd/thanos/config.go

```diff
@@ -161,3 +167,53 @@ func (wc *webConfig) registerFlag(cmd extkingpin.FlagClause) *webConfig {
 	cmd.Flag("web.disable-cors", "Whether to disable CORS headers to be set by Thanos. By default Thanos sets CORS headers to be allowed by all.").Default("false").BoolVar(&wc.disableCORS)
 	return wc
 }
+
+type queryConfig struct {
```

Comment:
Could we maybe put that next to the Querier for visibility? (: 

or is this shared across components?

Reply:
Ohh yeah, this struct actually parses `--query.*` flags (e.g `query.sd-interval`, `query.auto-downsampling`) so It is used by ruler and can later be re-used for querier (and other components that need it).

Thinking of it now, it would be easy to mistake it as a config struct for querier (similar to `ruleConfig` and `sidecarConfig`). Maybe it needs to be renamed? (not sure what a more appropriate name could be). Or it stays this way and querier's config struct could be something like `querierConfig`.

Reply:
Don't forget to run `make docs` at some point to fix CI issue (: